Enable inductor perf test on GCP A100

weiwangmeta opened the pull request:
This PR tries to enable inductor performance nightly testing on A100 runner provided by GCP. Currently these GCP runners were created and maintained using scripts in https://github.com/fairinternal/pytorch-gha-infra/pull/82. 
For some reason the artifacts cannot (and does not need to) be uploaded to S3, so adding use-gha parameter to _linux-test.yml to avoid creating a new but mostly identical _linux-test.yml. 

Workflow test results: https://github.com/pytorch/pytorch/actions/runs/3642340544/jobs/6149691109 

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

desertfire commented on .github/workflows/inductor-perf-test-nightly.yml:
```diff
@@ -0,0 +1,36 @@
+name: inductor
+
+on:
+  schedule:
```
How do we test this PR? I think adding a trigger with the `ciflow/nightly` label would be helpful here.

weiwangmeta replied:
Yes, I will add the ciflow/nightly trigger. Also I just realized the name conflict "inductor", I should rename it to inductor-perf-test.

desertfire commented on .jenkins/pytorch/test.sh:
```diff
@@ -274,6 +274,20 @@ test_inductor_huggingface() {
+  TEST_REPORTS_DIR=$(pwd)/test/test-reports
+  mkdir -p "$TEST_REPORTS_DIR"
+  # Check inference with --float32
+  python benchmarks/dynamo/huggingface.py --performance \
```
We are not monitoring the inference perf on our dashboard. This is nice to have if it doesn't add too much test time.

weiwangmeta replied:
Let me remove the inference perf part, as the total huggingface testing time seems long (1h30m).

weiwangmeta replied:
And I will also remove the inference perf part for timm and torchbench

desertfire commented on .jenkins/pytorch/test.sh:
```diff
@@ -274,6 +274,20 @@ test_inductor_huggingface() {
+  # Check training with --amp
+  python benchmarks/dynamo/huggingface.py --training --performance \
+    --device cuda --inductor --amp --output "$TEST_REPORTS_DIR"/inductor_training_huggingface.csv
+}
```
Please leave a comment to mark that we are not checking accuracy here yet.

weiwangmeta commented:
https://github.com/pytorch/pytorch/actions/runs/3642340544/jobs/6149691109#step:10:682 has BlenderbotForCausalLM OOM (**huggingface** model)
https://github.com/pytorch/pytorch/actions/runs/3642340544/jobs/6149691462#step:10:1337 hf_GPT2_large has OOM (**torchbench**)
https://github.com/pytorch/pytorch/actions/runs/3642340544/jobs/6149691462#step:10:1505 hf_T5 has OOM (**torchbench**)

weiwangmeta committed:
Enable inductor nightly perf test on GCP A100

weiwangmeta committed:
re-use timm shard as --total-partitions is hardcoded as 2 partitions for now

xuzhao9 commented:
I suggest we tune the GPU frequency like in:
```
https://github.com/pytorch/benchmark/blob/main/.github/workflows/userbenchmark-a100.yml#L37
```

weiwangmeta committed:
Increase the frequency to three times a day (UTC time)
1, 9, 15 corresponds to 5pm, 1am, 9am California time, 8pm, 4am, and 12pm Eastern time

desertfire commented on .jenkins/pytorch/test.sh:
```diff
@@ -284,6 +284,18 @@ test_inductor_huggingface() {
   test_inductor_benchmark huggingface
 }
 
+test_inductor_huggingface_perf() {
```
Some refactoring has been done upstream, https://github.com/pytorch/pytorch/blob/17941b12e0f0aeb018f5bb68502eca78bdf6f60b/.jenkins/pytorch/test.sh#L259. Please update correspondingly.

weiwangmeta replied:
Done, thanks for the reminder.

desertfire commented on .jenkins/pytorch/test.sh:
```diff
@@ -750,19 +788,39 @@ elif [[ "${TEST_CONFIG}" == *dynamo* && "${SHARD_NUMBER}" == 2 && $NUM_TEST_SHAR
   install_filelock
   install_triton
   test_dynamo_shard 2
+elif [[ "${TEST_CONFIG}" == *inductor_huggingface_perf* ]]; then
```
Can we merge this `elif` with the following on and use `if` only for the last step to select test_inductor_huggingface or test_inductor_huggingface_perf?

weiwangmeta replied:
Done, thanks for the suggestion.

weiwangmeta committed:
code refactoring

anijain2305 reviewed (approved):
Looks good to me. But, let's wait for @desertfire for final approval.

weiwangmeta commented:
@pytorchbot merge
